src/audio/iir.c

```diff
@@ -118,8 +118,12 @@ int32_t iir_df2t(struct iir_state_df2t *iir, int32_t x)
 			 */
 			acc = Q_SHIFT_RND(acc, 45 + iir->coef[c + 5], 31);
 			in = sat_int32(acc);
-			c += 7; /* Next coefficients section */
-			d += 2; /* Next biquad delays */
+
+			/* Proceed to next biquad coefficients and delay
+			 * lines.
+			 */
+			c += SOF_EQ_IIR_NBIQUAD_DF2T;
+			d += 2;
```

Comment:
magic number ?

Reply:
Yes sort of magic. It's a 2nd order filter so there are two delays per block but there's also biquads with more delays (direct form-I) so it's fair to fix this.